page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/data/DataParser.java

```diff
@@ -76,70 +73,43 @@ private Node replaceArgumentPlaceholders(Node argument) throws DataException {
             argument = replaceDataTablePlaceholders(dataTable);
         } else if (argument instanceof DocString) {
             DocString docString = (DocString) argument;
-            argument = new DocString(docString.getLocation(), docString.getContentType(), replaceStepPlaceholders(docString.getContent()));
+            argument = new DocString(docString.getLocation(), docString.getContentType(), replaceDataPlaceholders(docString.getContent()));
 
         }
         return argument;
     }
 
-    private String replaceStepPlaceholders(String raw) throws DataException {
-        Pattern stepDataPattern = Pattern.compile(STEP_PARSE_REGEX);
+    private String replaceDataPlaceholders(String raw) throws DataException {
+        Pattern stepDataPattern = Pattern.compile(PATH_PARSE_REGEX);
         Matcher stepDataMatcher = stepDataPattern.matcher(raw);
-        StringBuilder replacedStep = new StringBuilder(raw);
-        // Offset to handle diff between placeholder length and value length
-        int offset = 0;
+        StringBuffer replacedStep = new StringBuffer(raw);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder". ([squid:S1149](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1149))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=89&issues=AWW6t3U0_eEZPtUmV-5g&open=AWW6t3U0_eEZPtUmV-5g)

[ISSUE_KEY]: AWW6t3U0_eEZPtUmV-5g